resource.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 Tailscale Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build windows
+// +build windows
+
+package walk
+
+import (
+	"unsafe"
+
+	"github.com/dblohm7/wingoes/com"
+	"github.com/tailscale/win"
+	"golang.org/x/exp/constraints"
+	"golang.org/x/sys/windows"
+)
+
+// Resource encapsulates data loaded from the executable's Win32 resources.
+type Resource struct {
+	data []byte
+}
```

Comment:
```suggestion
// Resource encapsulates data loaded from the executable's Win32 resources.
type Resource []byte
```
?

If you'd like to encapsulate it, that's okay too. But then maybe update the functions below to return `(Resource, error)`?
There's probably no good reason to return a pointer to a slice, and this is essentially a value type.

Reply:
Actually, I think the functions should return `(Resource, error)` regardless of whether we make this a named slice type.